debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/ResumeTokens.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.mongodb;
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+
+import org.bson.BsonDocument;
+import org.bson.BsonTimestamp;
+import org.bson.BsonValue;
+
+/**
+ * Utilities for working with MongoDB resume tokens.
+ *
+ * Adaptation of <a href="https://github.com/mongodb/mongo-kafka/blob/master/src/main/java/com/mongodb/kafka/connect/util/ResumeTokenUtils.java">ResumeTokenUtils</a>
+ */
+public final class ResumeTokens {
+
+    public static BsonTimestamp getTimestamp(final BsonDocument resumeToken) {
+        BsonValue data = getData(resumeToken);
+        byte[] dataBytes = getBytes(data);
+        ByteBuffer dataBuffer = ByteBuffer.wrap(dataBytes).order(ByteOrder.BIG_ENDIAN);
+
+        // Cast to an int then remove the sign bit to get the unsigned value
+        int canonicalType = ((int) dataBuffer.get()) & 0xff;
+        if (canonicalType != 130) {
+            throw new IllegalArgumentException("Expected canonical type equal to 130, but found " + canonicalType);
+        }
+
+        long timestampAsLong = dataBuffer.asLongBuffer().get();
+        return new BsonTimestamp(timestampAsLong);
+    }
+
+    public static BsonValue getData(BsonDocument resumeToken) {
+        if (!resumeToken.containsKey("_data")) {
+            throw new IllegalArgumentException("Expected _data field in resume token");
+        }
+
+        return resumeToken.get("_data");
+    }
+
+    private static byte[] getBytes(BsonValue data) {
```

Comment:
This probably belongs to `MongoUtils`

Reply:
@jpechane  disagree. MongoUtils is slowly becoming a mess of about bilion semi related utility methods. I'd like to keep it here at least until we need it also somewhere else.

Reply:
Then let's make a separate util class for it like `BsonUtil`.

Reply:
At closer look this method really makes sense only in context of the `_data` field from resume token. Making it public for anything else than this class doesn't really make sense. Though I would rename it to `getDataByte` or something similar.

Reply:
> Making it public for anything else than this class doesn't really make sense

This method is private, does that change your opinion?

> Though I would rename it to getDataByte or something similar.

Makes sense.

Reply:
> @jpechane disagree. MongoUtils is slowly becoming a mess of about bilion semi related utility methods. I'd like to keep it here at least until we need it also somewhere else.

Imho, I agree with @jcechace here. In my experience, `<X>Utils` tend to become a broad dumping ground that only shares the weakest form of cohesion: [coincidental](https://en.wikipedia.org/wiki/Cohesion_(computer_science)#Types_of_cohesion). While just a convention, the more modern `<X>s` style promoted by libraries such as Guava and the JDK tend to be more focused and akin to "companion classes" in other languages like Scala or Kotlin.

That said, happy to follow the `<X>Utils` convention if that is more common in the code base. Though I would rather not widen scope to `BsonUtils` for the same reasons discussed above.  There is opportunity for reuse / encapsulate in other call sites as well. For example:
- https://github.com/debezium/debezium/blob/fcb0e77bbb0a3f5c7873b89e066f20d08dfc2949/debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/MongoDbStreamingChangeEventSource.java#L224-L226
- https://github.com/debezium/debezium/blob/fcb0e77bbb0a3f5c7873b89e066f20d08dfc2949/debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/SourceInfo.java#L308